device_response_writer.go

```diff
@@ -0,0 +1,21 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package fosite
+
+import (
+	"context"
+)
+
+func (f *Fosite) NewDeviceResponse(ctx context.Context, r DeviceRequester, session Session) (DeviceResponder, error) {
+	var resp = &DeviceResponse{}
+
+	r.SetSession(session)
+	for _, h := range f.Config.GetDeviceEndpointHandlers(ctx) {
```

Comment:
this is a common pattern in fosite, the reasoning behind this is that these handlers are complementary to one another. 

For example for the token handler, you would first have a handler that handles oauth2 and then you would have a handler that implements oidc (e.g. adding id tokens). TBH I doubt anyone is going to use extra handlers for this endpoint, but I think it's best to keep it consistent (also you never know).

Reply:
hmmm.. ok.. i still think it's a weird pattern.. and possibly wrong..

Reply:
Why wrong? It is common to provide a way to the caller to extend the features of the library and in some cases it cannot be avoided (see OIDC on authorization endpoint). Perhaps it would be better if they didn't call them handlers but middlewares or plugins. Is there an problematic case you see?

Reply:
because the intent of this logic is unclear as i mentioned in the first comment.. once the first handler returns an error, none of the other handlers will be used.. so.. what is the purpose of this?